hmda/src/test/scala/hmda/validation/rules/ts/validity/V607Spec.scala

```diff
@@ -0,0 +1,17 @@
+package hmda.validation.rules.ts.validity
+
+import hmda.model.filing.ts.TransmittalSheet
+import hmda.model.filing.ts.TsGenerators._
+import hmda.validation.rules.EditCheck
+import hmda.validation.rules.ts.TsEditCheckSpec
+
+class V607Spec extends TsEditCheckSpec {
+  override def check: EditCheck[TransmittalSheet] = V607
+
+  property("Tax ID must be a valid format") {
+    forAll(tsGen) { ts =>
+      ts.mustPass
+      ts.copy(taxId = "").mustFail
```

Comment:
This test is very shallow, only tests that taxId == "" fails. This is fine if we have more tests in `PredicateRegexSpec` that exercise failures for the `taxId` regular expression. See examples of malformed emails, phone numbers etc. in `PredicateRegexSpec` for more test coverage